src/TesApi.Web/BatchPool.cs

```diff
@@ -0,0 +1,584 @@
+
+            switch (exceptions.Count)
+            {
+                case 0:
+                    return;
+
+                case 1:
+                    throw exceptions.First();
+
+                default:
+                    throw new AggregateException(exceptions.SelectMany(Flatten).ToArray());
+            }
+
+            static IEnumerable<Exception> Flatten(Exception ex)
+                => ex switch
+                {
+                    AggregateException aggregateException => aggregateException.InnerExceptions,
+                    _ => Enumerable.Empty<Exception>().Append(ex),
+                };
+
+            async ValueTask PerformTask(ValueTask serviceAction, CancellationToken cancellationToken)
+            {
+                if (!cancellationToken.IsCancellationRequested)
+                {
+                    try
+                    {
+                        await serviceAction;
+                    }
+                    catch (Exception ex)
+                    {
+                        exceptions.Add(ex);
+                        RemoveMissingPools(ex);
+                    }
+                }
+            }
+
+            void RemoveMissingPools(Exception ex)
+            {
+                switch (ex)
+                {
+                    case AggregateException aggregateException:
+                        foreach (var e in aggregateException.InnerExceptions)
+                        {
+                            RemoveMissingPools(e);
+                        }
+                        break;
+
+                    case BatchException batchException:
+                        if (batchException.RequestInformation.BatchError.Code == BatchErrorCodeStrings.PoolNotFound)
+                        {
+                            _logger.LogError(ex, "Batch pool {PoolId} is missing. Removing it from TES's active pool list.", Pool.PoolId);
+                            _ = _batchPools.RemovePoolFromList(this);
+                        }
+                        break;
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public async ValueTask<DateTime> GetAllocationStateTransitionTime(CancellationToken cancellationToken = default) // TODO: put this at front of list by returning earliest possible time, or put at end of list by returning UtcNow?
+            => (await _azureProxy.GetBatchPoolAsync(Pool.PoolId, new ODATADetailLevel { SelectClause = "allocationStateTransitionTime" }, cancellationToken)).AllocationStateTransitionTime ?? DateTime.UtcNow;
+
+        /// <inheritdoc/>
+        public async ValueTask CreatePoolAndJobAsync(Microsoft.Azure.Management.Batch.Models.Pool poolModel, bool isPreemptible, CancellationToken cancellationToken)
+        {
+            try
+            {
+                CloudPool pool = default;
+                var poolInfo = await _azureProxy.CreateBatchPoolAsync(poolModel, isPreemptible);
+                await Task.WhenAll(
```

Comment:
Out of the three operations (create the pool in the account, retrieve the `CloudPool` object representing that newly created pool (to access two items of metadata), and create the job, the only two that need to be ordered are creating the pool and retrieving the SDK object. The job, while here not being created until after the pool is created, is actually created after the pool. We just don't want to add this object to `BatchScheduler`'s pool list until everything is created and configured.

So, we do want to run those two tasks in parallel. We could make creating and obtaining the SDK object one "task" and creating the job the other parallel task, but then we wouldn't have the convenience of using a PoolInformation object to "ensure" that we're creating a job for that pool instead of some arbitrary string that should represent a pool, so this seems to be the right shape.